File: sample/src/main/java/com/afollestad/dragselectrecyclerviewsample/RectangleView.java
package com.afollestad.dragselectrecyclerviewsample;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;

/** @author Aidan Follestad (afollestad) */
public class RectangleView extends FrameLayout {

  public RectangleView(Context context) {
    super(context);
  }

  public RectangleView(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  public RectangleView(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
  }

  @Override
  protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
    super.onMeasure(widthMeasureSpec, heightMeasureSpec);
    setMeasuredDimension(getMeasuredWidth(), (int) (getMeasuredWidth() * 1.4f));
  }
}

File: sample/src/main/java/com/afollestad/dragselectrecyclerviewsample/RectangleView.kt
/*
 * Licensed under Apache-2.0
 *
 * Designed and developed by Aidan Follestad (@afollestad)
 */
package com.afollestad.dragselectrecyclerviewsample

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout

/** @author Aidan Follestad (afollestad) */
class RectangleView(
  context: Context,
  attrs: AttributeSet?
) : FrameLayout(context, attrs) {

  override fun onMeasure(
    widthMeasureSpec: Int,
    heightMeasureSpec: Int
  ) {
    super.onMeasure(widthMeasureSpec, heightMeasureSpec)
    setMeasuredDimension(measuredWidth, (measuredWidth * 1.4f).toInt())
  }
}
